src/aiq/utils/io/yaml_tools.py

```diff
@@ -14,37 +14,128 @@
 # limitations under the License.
 
 import io
+import os
+import re
 import typing
 
 import yaml
 
 from aiq.utils.type_utils import StrPath
 
 
+def _interpolate_variables(value: str | int | float | bool | None) -> str | int | float | bool | None:
+    """
+    Interpolate variables in a string with the format ${VAR:-default_value}.
+    If the variable is not set, the default value will be used.
+    If no default value is provided, an empty string will be used.
+
+    Args:
+        value (str | int | float | bool | None): The value to interpolate variables in.
+
+    Returns:
+        str | int | float | bool | None: The value with variables interpolated.
+    """
+
+    if not isinstance(value, str):
+        return value
+
+    def replace_var(match):
+        var_name = match.group(1)
+        default_value = match.group(2) if match.group(2) else ""
+        return os.environ.get(var_name, default_value)
+
+    pattern = r'\${([^:}]+)(?::-([^}]*))?}'
+    return re.sub(pattern, replace_var, value)
```

Comment:
What about using something like [`expandvars`](https://pypi.org/project/expandvars/) so we dont have to maintain this logic ourselves? There are more edge cases which need to be dealt with (escape sequences, default values with quotes, etc).